bundles/persistence/org.openhab.persistence.influxdb08/java/org/openhab/persistence/influxdb08/internal/InfluxDBPersistenceService.java

```diff
@@ -221,8 +221,6 @@ public void store(Item item, String alias) {
     State state = null;
     if (item instanceof DimmerItem || item instanceof RollershutterItem) {
       state = item.getStateAs(PercentType.class);
-    } else if (item instanceof ColorItem) {
```

Comment:
Explain or revert all the changes to this file.

Reply:
Here `ColorItem` is a subclass of `DimmerItem` so again the check is redundant and there is dead code. I have removed the dead code and the redundant check, but again an alternative solution would be to check for `ColorItem` before checking for `DimmerItem`. Which is better here ? Similarly, there are checks in this class for `HSBType`, which are done after checks for its superclass. Again, I've removed the redundant code, but could instead move the check for `HSBType` earlier if you prefer that .